src/mono/wasm/runtime/marshal-to-cs.ts

```diff
@@ -305,10 +308,17 @@ function _marshal_task_to_cs(arg: JSMarshalerArgument, value: Promise<any>, _?:
     const holder = new TaskCallbackHolder(value);
     setup_managed_proxy(holder, gc_handle);
 
+    if (monoWasmThreads)
+        addUnsettledPromise();
```

Comment:
I'm not sure what we should do when we return a Task to a JS worker.

I'm having trouble imagining what a sample would look like that does this that doesn't also somehow return to C#.  (ie: I first `JSImport` something and that something is async and it does an `await` on a `JSExport` - so everything is alive as long as the c# is awaiting the JSImport-ed thing)

Reply:
Yea, in fact I tried something like that with setTimeout. 

I'm not sure if it's better to add something when we pass a js handle for a c# object, or just to add a public c# API to ensure the thread stays alive. It seems hard to guess what is the right thing.

Reply:
The other type of "event" which should keep the thread alive is events from WebSocket. 
I guess JS side of the WS code could call internal `addUnsettledPromise()`.

Perhaps we need better name as it's not only about promises.

Reply:
> And so you would have to keep the thread alive until the last C# function proxy was GC'd by `FinalizationRegistry`.

I think this is probably what we need to do, then.  Both for C# functions and C# Tasks.  Maybe even all C# objects that get passed to JS.

If something in JS is holding on to a C# object of any kind, the C# threadpool worker has to stay alive because JS may trigger some C# Task to become unblocked and require executing on that webworker.

Reply:
@pavelsavara So did we decide something here?  Should I block the worker thread from exiting while it has any proxies that haven't been GC'd on the JS side?

Reply:
To really free all proxies, worst case scenarios could involve multiple runs of both GCs. And I don't like it.

So, probably not on this PR. Let's create issue for it and link it from this place in code where we only count promises.
So that we have chance to experiment with different solutions separately and unblock the rest of the progress on MT.

Reply:
Created #85052